crates/pgmq/src/lib.rs

```diff
@@ -131,15 +131,17 @@ impl PGMQueue {
 
     /// Create a queue
     pub async fn create(&self, queue_name: &str) -> Result<(), errors::PgmqError> {
-        let create = query::create(queue_name);
-        let index: String = query::create_index(queue_name);
-        sqlx::query(&create).execute(&self.connection).await?;
-        sqlx::query(&index).execute(&self.connection).await?;
+        let mut tx = self.connection.begin().await?;
+        let setup = query::init_queue(queue_name);
+        for q in setup {
+            sqlx::query(&q).execute(&mut tx).await?;
+        }
+        tx.commit().await?;
```

Comment:
creating a queue now does a bunch of things, all are no-ops if the data already exists. metadata table, queue table, queue archive table, create index, insert metadata to queue metadata table